deployment/src/main/java/io/quarkiverse/qute/web/deployment/QuteWebProcessor.java

```diff
@@ -60,24 +61,28 @@ public void collectTemplatePaths(TemplateFilePathsBuildItem templateFilePaths,
                 continue;
             }
             LOG.debugf("Web template found: %s", path);
-            paths.produce(new QuteWebTemplatePathBuildItem(path));
+            paths.produce(new QuteWebTemplateBuildItem(path, null));
         }
     }
 
     @BuildStep
     @Record(ExecutionTime.RUNTIME_INIT)
     @Consume(SyntheticBeansRuntimeInitBuildItem.class)
-    public RouteBuildItem produceTemplatesRoute(QuteWebRecorder recorder, List<QuteWebTemplatePathBuildItem> templatePaths,
+    public RouteBuildItem produceTemplatesRoute(QuteWebRecorder recorder, List<QuteWebTemplateBuildItem> templates,
             HttpRootPathBuildItem httpRootPath, QuteWebBuildTimeConfig config) {
-        if (templatePaths.isEmpty()) {
+        if (templates.isEmpty()) {
             // There are no templates to serve
             return null;
         }
+        final var templateLinks = templates.stream().filter(t -> !Objects.isNull(t.link()))
```

Comment:
It might be better to add `QuteWebTemplateBuildItem#hasLink()` so that you can write `templates.stream().filter(QuteWebTemplateBuildItem::hasLink).collect(...)`. It's definitely easier to read...

Reply:
done